src/components/Tool/Appeal/Modals/PledgeModal/PledgeModal.test.tsx

```diff
@@ -26,20 +31,58 @@ const router = {
   query: { accountListId },
   isReady: true,
 };
+const defaultPledge = {
+  id: 'pledgeId',
+  amount: 110,
+  amountCurrency: 'USD',
+  appeal: {
+    id: appealId,
+  },
+  expectedDate: '2024-08-08',
+  status: PledgeStatusEnum.NotReceived,
+};
 const handleClose = jest.fn();
 const mutationSpy = jest.fn();
+const mockEnqueue = jest.fn();
+jest.mock('notistack', () => ({
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  ...jest.requireActual('notistack'),
+  useSnackbar: () => {
+    return {
+      enqueueSnackbar: mockEnqueue,
+    };
+  },
+}));
 
 interface ComponentsProps {
   pledge?: AppealContactInfoFragment['pledges'][0];
+  selectedAppealStatus?: AppealStatusEnum;
+  updateAccountListPledge?: UpdateAccountListPledgeMutation;
+  createAccountListPledge?: CreateAccountListPledgeMutation;
 }
 
-const Components = ({ pledge = undefined }: ComponentsProps) => (
+const Components = ({
+  pledge = undefined,
+  selectedAppealStatus,
+  updateAccountListPledge = {},
+  createAccountListPledge = {},
+}: ComponentsProps) => (
   <I18nextProvider i18n={i18n}>
     <LocalizationProvider dateAdapter={AdapterLuxon}>
       <SnackbarProvider>
         <ThemeProvider theme={theme}>
           <TestRouter router={router}>
-            <GqlMockedProvider onCall={mutationSpy}>
+            <GqlMockedProvider<{
+              UpdateAccountListPledge: UpdateAccountListPledgeMutation;
+              CreateAccountListPledge: CreateAccountListPledgeMutation;
+            }>
+              onCall={mutationSpy}
+              mocks={{
+                UpdateAccountListPledge: updateAccountListPledge,
+                CreateAccountListPledge: createAccountListPledge,
```

Comment:
If you changed to this, the render functions wouldn't have to do `updateAccountListPledge={{ updateAccountListPledge: { ... } }}` and could just do `updateAccountListPledge={{ ... }}`

```suggestion
                UpdateAccountListPledge: { updateAccountListPledge },
                CreateAccountListPledge: { createAccountListPledge },
```